MediaGallerySynchronizationApi/Api/SelectByBatchesGeneratorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronizationApi\Api;
+
+/**
+ * Select data from database by provided batch size
+ */
+interface SelectByBatchesGeneratorInterface
```

Comment:
Naming improvement suggestion
```suggestion
interface FetchBatchesInterface
```